src/main/java/org/codehaus/plexus/util/xml/Xpp3Dom.java

```diff
@@ -244,19 +253,61 @@ public void addChild( Xpp3Dom xpp3Dom )
         }
         else
         {
-            ArrayList<Xpp3Dom> children = new ArrayList<Xpp3Dom>();
-            int size = childList.size();
+            ArrayList<Xpp3Dom> children = null;
+
+            for ( Xpp3Dom configuration : childList )
+            {
+                if ( name.equals( configuration.getName() ) )
+                {
+                    if ( children == null )
+                    {
+                        children = new ArrayList<Xpp3Dom>();
+                    }
+                    children.add( configuration );
+                }
+            }
+
+            if ( children != null )
+            {
+                return children.toArray( EMPTY_DOM_ARRAY );
+            }
+            else
+            {
+                return EMPTY_DOM_ARRAY;
+            }
+        }
+    }
+
+    public List<Xpp3Dom> getChildrenAsList( String name )
```

Comment:
Why is this method necessary? why not simply say `Arrays#asList()`? This is duplicate code.

Reply:
The goal is to avoid creating an array from the list, then wrapping the list in `Arrays.asList()` just after in the `mergeIntoXpp3Dom`.  I can rewrite the `getChildren` to be `getChildrenAsList(xx).toArray( EMPTY_DOM_ARRAY )` to avoid the duplication.

Reply:
And I can make it private too.